core/src/main/java/com/google/instrumentation/trace/TraceExporter.java

```diff
@@ -54,6 +71,352 @@
+        LatencyBasedSampledSpansFilter filter);
+
+    /**
+     * Returns a list of failed spans (spans with {@link Status} different than {@link Status#OK})
+     * that are matching the {@code filter}.
+     *
+     * <p>Error based sampled spans are available only for span names registered using {@link
+     * #collectSamplesForSpanNames(List)}.
+     *
+     * @param filter used to filter the returned sampled spans.
+     * @return a list of failed spans that are matching the {@code filter}.
+     */
+    public abstract List<SpanData> getErrorBasedSampledSpans(ErrorBasedSampledSpansFilter filter);
+
+    /**
+     * Registers a list of span names for which the library will collect latency based sampled spans
+     * and error based sampled spans.
+     *
+     * @param spanNames list of span names for which the library will collect samples.
+     */
+    public abstract void collectSamplesForSpanNames(List<String> spanNames);
+
+    /** The summary of all in-process debugging information. */
+    @AutoValue
+    @Immutable
+    public abstract static class Summary {
+
+      Summary() {}
+
+      /**
+       * Returns a map with all the span names and the summary of available data for each different
+       * span name.
+       *
+       * @return a map with all the span names and the summary.
+       */
+      public abstract Map<String, PerSpanNameSummary> getPerSpanNameSummary();
+
+      /** Summary of all available data for a span name. */
+      @AutoValue
+      @Immutable
+      public abstract static class PerSpanNameSummary {
+
+        PerSpanNameSummary() {}
+
+        /**
+         * Returns the number of active spans.
+         *
+         * @return the number of active spans.
+         */
+        public abstract int getNumberOfActiveSpans();
+
+        /**
+         * Returns the list of all latency based sampled buckets summary.
+         *
+         * @return the list of all latency based sampled buckets summary.
+         */
+        public abstract List<LatencyBucketSummary> getLatencyBucketSummaries();
+
+        /**
+         * Returns the list of all error based sampled buckets summary.
+         *
+         * @return the list of all error based sampled buckets summary.
+         */
+        public abstract List<ErrorBucketSummary> getErrorBucketSummaries();
+
+        /**
+         * Summary of a latency based sampled spans bucket. Contains {code Span} samples with
+         * latency between [latencyLowerNs, latencyUpperNs).
+         */
+        @AutoValue
+        @Immutable
+        public abstract static class LatencyBucketSummary {
+
+          LatencyBucketSummary() {}
+
+          /**
+           * Returns a new instance of {@code LatencyBucketSummary}.
+           *
+           * @param numSamples the number of sampled spans.
+           * @param latencyLowerNs the latency lower bound.
+           * @param latencyUpperNs the latency upper bound.
```

Comment:
I think your change made it clearer.  I figured that since there is already a method Javadoc with complete param tags, the Javadoc should probably give a full description of each parameter, even if some of the information is mentioned elsewhere.